src/SFA.DAS.EmployerAccounts.Web/Controllers/EmployerAgreementController.cs

```diff
@@ -240,63 +240,46 @@ public async Task<ActionResult> GetSignedPdfAgreement(string agreementId, string
         }
 
         [HttpGet]
-        [Route("agreements/remove")]
-        public async Task<ActionResult> GetOrganisationsToRemove(string hashedAccountId)
+        [Route("agreements/{accountLegalEntityHashedId}/remove")]
+        public async Task<ActionResult> ConfirmRemoveOrganisation(string accountLegalEntityHashedId, string hashedAccountId)
         {
-            var model = await _orchestrator.GetLegalAgreementsToRemove(hashedAccountId, OwinWrapper.GetClaimValue(ControllerConstants.UserRefClaimKeyName));
-            return View(model);
+            var model = await _orchestrator.GetConfirmRemoveOrganisationViewModel(accountLegalEntityHashedId, hashedAccountId, OwinWrapper.GetClaimValue(ControllerConstants.UserRefClaimKeyName));
+
+            return View(model.Data.CanBeRemoved ? ControllerConstants.ConfirmRemoveOrganisationActionName : ControllerConstants.CannotRemoveOrganisationViewName, model.Data);
         }
 
-        [HttpGet]
-        [Route("agreements/remove/{agreementId}")]
-        public async Task<ActionResult> ConfirmRemoveOrganisation(string agreementId, string hashedAccountId)
+        [HttpPost]
+        [Route("agreements/{accountLegalEntityHashedId}/remove")]
```

Comment:
Shouldn't the route be?

accounts/{HashedAccountId}/organisations/{accountLegalEntityHashedId}/agreements/remove ?

Reply:
No, potentially it should be in a different controller now as the remove is done by account legal entity and not agreement.